### ISSUE TITLE
Declaring a state's parent using `parent:` and `name: 'parent.child'` should Error

### ISSUE THREAD

**Issue by christopherthielen** [angular-ui/ui-router#1579]
See comments in 5797c2e

**Comment by anotherpit** [comment]
Should it?

This first snippet looks pretty legitimate, doesn't it? And it also works as expected.
```
$stateProvider
    .state("List", {
        url: "/list"
    })
    .state("Item", {
        url: "/{id:string}",
        parent: "List"
    });
```

The second one is pretty similar:
```
$stateProvider
    .state("Smth.List", {
        url: "/list"
    })
    .state("Smth.Item", {
        url: "/{id:string}",
        parent: "Smth.List"
    });
```
But it fails with `TypeError: Cannot read property 'navigable' of undefined`.

I suggest not to force errors, but instead stick to some consistent precedence. Currently, [`stateBuilder.parent()`](https://github.com/angular-ui/ui-router/blob/master/src/state.js#L33) prefers `parent` over dot notation (which makes more sense to me), while [`registerState()`](https://github.com/angular-ui/ui-router/blob/master/src/state.js#L178) gives more respect to complex names. Looks like it's enough to fix only one these methods to fix the issue.

**Comment by nateabele** [comment]
> Should it?

Yes. There's no way to know which should take precedence by looking at the code. Ambiguous designs are almost always a bad idea.

**Comment by christopherthielen** [comment]
Closed by b4621f3

**Comment by cserajeevdas** [comment]
I'm getting a similar error. If the state name is not intended to be parent-child, will the renaming and removing dot(.) will solve the issue? What else is the workaround?
